addons/functions/functions/fnc_hitPart.sqf

```diff
@@ -0,0 +1,34 @@
+#include "script_component.hpp"
+/*
+ * Author: nomisum
+ * Server receives hitPart Events from Clients and calulates
+ *
+ * Arguments:
+ * 0: Trench <OBJECT>
+ * 1: Splash Damage <NUMBER>
+ *
+ * Return Value:
+ * NONE
+ *
+ * Example:
+ * [TrenchObj, 5] call grad_trenches_functions_fnc_hitPart;
+ *
+ * Public: No
+ */
+
+
+params ["_trench", "_position", "_splashDamage"];
+
+// send fx to clients
+[QGVAR(hitFX), [vectorDir _trench, ASLToAGL (_position)]] call CBA_fnc_globalEvent;
+
+private _progress = _trench getVariable ["ace_trenches_progress", 0];
```

Comment:
As far as I can see, you mostly read ace_trench_progress to then add or subtract something from it.

So another suggestion for refactor: 

create new `FUNC(addTrenchProgress)` that contains:

```sqf
params ["_trench", "_addedProgress", "_time"];
private _newProgress = (_trench getVariable["ace_trenches_progress", 0]) + _addedProgress;
if (_newProgress > 0) then {
    [_trench, _newProgress, _time] call FUNC(setTrenchProgress);
} else {
    [_trench] call FUNC(deleteTrench);
};
```
and use that whereever progress is changed :thinking: , so the manipulation of ace_trenches_progress is reduced to fewer places. hmmm. optional though, I think there is a lot of other places that would need change as well.